modules/check_in/app/controllers/check_in/v2/demographics_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module CheckIn
+  module V2
+    class DemographicsController < CheckIn::ApplicationController
+      before_action :before_logger, only: %i[update], if: :additional_logging?
+      after_action :after_logger, only: %i[update], if: :additional_logging?
+
+      def update
+        head :not_implemented
+      end
+
+      def permitted_params
+        params.require(:demographics).permit(:demographic_confirmations).tap do |demographics|
+          demographics.require(:demographics_up_to_date, :next_of_kin_up_to_date, :emergency_contact_up_to_date)
```

Comment:
Is this for the nested params under `demographic_confirmations`? Can we not do:
```suggestion
        params.require(:demographics).permit(demographic_confirmations: %i[
                                               demographics_up_to_date
                                               next_of_kin_up_to_date
                                               emergency_contact_up_to_date
                                             ])
```

https://guides.rubyonrails.org/v6.1/action_controller_overview.html#nested-parameters

Reply:
Thanks for the suggestion.  

There is an issue with both the approaches where `demographic_confirmations` not restricted with all three attributes nested under it.  

I have changes in a different branch and planning to open a separate PR with these changes including rspecs.  Please let me know your opinion.

Reply:
What is the incoming request format?

Reply:
Downstream service CHIP accepts,
```
{
    "demographicConfirmations": {
        "demographicsConfirmedAt": "2022-02-04T12:00:00Z",
        "demographicsNeedsUpdate": false,
        "emergencyContactConfirmedAt": "2022-02-04T12:00:00Z",
        "emergencyContactNeedsUpdate": false,
        "nextOfKinConfirmedAt": "2022-01-01T12:00:00Z",
        "nextOfKinNeedsUpdate": true
    },
    "patientDfn": "418",
    "stationNo": "500"
}
```

vets-api demographics controller will accept `demographics_confirmation` as an optional parameter and if it exists, then `demographicsNeedsUpdate`, `emergencyContactNeedsUpdate` & `nextOfKinNeedsUpdate` will become mandatory.

Request will look like
```
{
  "demographics" : {
    "demographicConfirmations": {
        "demographicsNeedsUpdate": false,
        "emergencyContactNeedsUpdate": false,
        "nextOfKinNeedsUpdate": true
    }
  }
}
```

Reply:
So with that, I don't think we should validate the 3 keys under `demographicConfirmations`, since we'll probably be parsing this JSON into a model anyways. So maybe just:
`params.require(:demographics).permit(:demographic_confirmations)`

Reply:
As discussed offline, will include the validation for attributes under `demographicConfirmations` in next PR